opengever/officeconnector/testing.py

```
@@ -0,0 +1,478 @@
+            document,
+            headers=headers,
+            view=payload.get('checkout'),
+            send_authenticator=True,
+            )
+
+        self.login(current_user, browser)
+        self.assertEquals(current_user.id, document.checked_out_by())
+        self.assert_journal_entry(
+            document,
+            DOCUMENT_CHECKED_OUT,
+            u'Document checked out',
+            )
+
+    def lock_document(self, browser, tokens, document):
+        """Logs out, uses JWT to WebDAV lock the document and logs back in."""
+        lock_manager = getMultiAdapter(
+            (document, self.request),
+            name='plone_lock_info',
+            )
+        self.assertFalse(lock_manager.is_locked())
+
+        current_user = api.user.get_current()
+        browser.logout()
+
+        browser.raise_http_errors = False
+
+        headers = {
+            'Authorization': ' '.join(('Bearer', tokens.get('raw_token'), )),
+            'Content-Type': 'text/xml; charset="utf-8"',
+            'Depth': '0',
+            'Timeout': 'Infinite, Second-4100000000',
+            }
+
+        data = (
+            '<?xml version="1.0" encoding="utf-8"?>\n'
+            '<d:lockinfo xmlns:d="DAV:">\n'
+            '  <d:lockscope><d:exclusive/></d:lockscope>\n'
+            '  <d:locktype><d:write/></d:locktype>\n'
+            '  <d:owner>\n'
+            '  <d:href>Office Connector</d:href>\n'
+            '  </d:owner>\n'
+            '</d:lockinfo>'
+            )
+
+        browser.webdav('LOCK', document, headers=headers, data=data)
+        self.assertEquals(200, browser.status_code)
+        self.assertTrue(lock_manager.is_locked())
+
+        try:
+            lock_token = ET.fromstring(browser.contents).find(
+                './d:lockdiscovery/d:activelock/d:locktoken/',
+                {'d': 'DAV:'},
+                ).text
+        except ET.ParseError:
+            lock_token = None
+        self.assertTrue(lock_token)
```

You are converting an `ET.ParseError` to an `AssertionError` here: the assertion error will always be raised when an `ET.ParseError` is raised. But why catch the `ET.ParseError` at all?
Now you remove information, so that you only have the info `AssertionError: None is not true` when it fails.

I would not catch the `ET.ParseError` at all and let it bubble up to the test output.
For example:
```python
self.assertTrue(ET.fromstring(...))
```